cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/adapter/AdapterServiceTests.java

```diff
@@ -157,7 +172,56 @@ public void testWorkerAdapter() throws Exception {
   }
 
   @Test
-  public void testAdapters() throws Exception {
+  public void testBatchRunRecords() throws Exception {
```

Comment:
this test runs for a long time and is very reliant on timing, which makes it a very fragile test.  I think it's fine, we don't need to test this and can rely on existing run records and lifecycle tests.